pybamm/expression_tree/unary_operators.py

```diff
@@ -112,6 +112,48 @@ def __init__(self, child):
         super().__init__("div", child)
 
 
+class SpatialAverage(SpatialOperator):
+    """A node in the expression tree representing a spatially averaging operator.
```

Comment:
I think this needs more documentation. I would normally assume that a spatially averaging would take place over the entire vector, but here the averaging is taking place over adjacent elements only.